core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/convert.kt

```diff
@@ -125,6 +126,48 @@ public fun <T : Any> DataColumn<T?>.convertToString(): DataColumn<String?> = con
 public fun <T : Any> DataColumn<T>.convertToDouble(): DataColumn<Double> = convertTo()
 public fun <T : Any> DataColumn<T?>.convertToDouble(): DataColumn<Double?> = convertTo()
 
+/**
+ * Parse String column to Double considering locale (number format).
+ * If [locale] parameter is defined, it's number format is used for parsing.
+ * If [locale] parameter is null, the current system locale is used. If column can not be parsed, then POSIX format is used.
+ */
+@JvmName("convertToDoubleFromString")
+public fun DataColumn<String>.convertToDouble(locale: Locale? = null): DataColumn<Double> {
+    if (locale is Locale) {
+        val explicitConverter = Parsers.getDoubleConverter(locale) as (String) -> Double?
+        return map { explicitConverter(it.trim()) ?: error("Can't convert `$it` to Double") }
+    } else {
+        return try {
+            val defaultConverter = Parsers.getDoubleConverter() as (String) -> Double?
+            map { defaultConverter(it.trim()) ?: error("Can't convert `$it` to Double") }
+        } catch (e: TypeConversionException) {
+            val posixConverter = Parsers.getDoubleConverter(Locale.forLanguageTag("C.UTF-8")) as (String) -> Double?
+            map { posixConverter(it.trim()) ?: error("Can't convert `$it` to Double") }
+        }
+    }
+}
+
+/**
+ * Parse String column to Double considering locale (number format).
+ * If [locale] parameter is defined, it's number format is used for parsing.
+ * If [locale] parameter is null, the current system locale is used. If column can not be parsed, then POSIX format is used.
+ */
+@JvmName("convertToDoubleFromStringNullable")
+public fun DataColumn<String?>.convertToDouble(locale: Locale? = null): DataColumn<Double?> {
+    if (locale is Locale) {
```

Comment:
What do you think? 
```kotlin    
val converter = if (locale != null) {
    Parsers.getDoubleConverter(locale)
} else {
    Parsers.getDoubleConverter()
}
```

Reply:
And then simply call `map { converter(it) }`? This it doesn't seem to work because we wanted the scalar converter not to use fallback logic. And it is implemented here now…